blocks/graph.py

```diff
@@ -520,3 +521,82 @@ def apply_dropout(computation_graph, variables, drop_prob, rng=None,
         replacement.tag.replacement_of = variable
 
     return computation_graph.replace(replacements)
+
+
+def apply_batch_normalization(computation_graph, variables, gammas,
+                              betas, axis=0, epsilon=1e-7):
+    """Returns a graph to variables in a computational graph.
+
+    Parameters
+    ----------
+    computation_graph : instance of :class:`ComputationGraph`
+        The computation graph.
+    variables : list of :class:`~tensor.TensorVariable`
```

Comment:
I guess my first question is, is there a simple way to infer which variables in a graph need to be batch-normalized? Even if we don't do it automagically, some guidance as to this point might be worth putting in the docs.